### PR TITLE
[SOS] [Linux] Add ELF.h and Dwarf.h from LLVM for gdbjit feature implementation

### PR DESCRIPTION
**Pull request by lucenticus**
This pull request adds ELF and Dwarf headers from LLVM compiler infrastructure. These files are needed for PR https://github.com/dotnet/coreclr/pull/6278 to generate elf & dwarf files for jitted code.
@jkotas, @janvorli PTAL
 
CC: @Dmitri-Botcharnikov @chunseoklee @seanshpark

### CODE REVIEW

File: THIRD-PARTY-NOTICES [hunk 1]
```diff
@@ -39,3 +39,49 @@ TASK FORCE DISCLAIMS ALL WARRANTIES, EXPRESS OR IMPLIED, INCLUDING
 BUT NOT LIMITED TO ANY WARRANTY THAT THE USE OF THE INFORMATION
 HEREIN WILL NOT INFRINGE ANY RIGHTS OR ANY IMPLIED WARRANTIES OF
 MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE.
+
+License notice for The LLVM Compiler Infrastructure
+---------------------------
```
Comment (jkotas) [inline THIRD-PARTY-NOTICES]: Nit - could you please make the line of `----` to be the same length as the previous line?

File: src/inc/Dwarf.h [hunk 2]
```diff
@@ -0,0 +1,711 @@
...
+//
+// For details on the Dwarf specfication see the latest DWARF Debugging
+// Information Format standard document on http://www.dwarfstd.org. This
+// file often includes support for non-released standard features.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_SUPPORT_DWARF_H
+#define LLVM_SUPPORT_DWARF_H
+
```
Comment (jkotas) [inline src/inc/Dwarf.h]: Could you please add verbatim copies of these files in this PR, and make any changes to them in separate PR - so that we have a record of what was changed.

File: src/inc/Dwarf.def [hunk 3]
```diff
@@ -0,0 +1,393 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (jkotas) [inline src/inc/Dwarf.def]: Could you please add these files into a dedicate directory (e.g. 'src\inc\llvm')

**Comment by jkotas** [comment]
It would be nice to include commit hash or date at which the files were taken.

File: src/inc/Dwarf.h (hunk 2 above)
Comment (lucenticus) [inline src/inc/Dwarf.h]: Do I need to add .NET license headers into these verbatim copies or add them in separate PR?

Comment (jkotas) [inline src/inc/Dwarf.h]: I do not think .NET license header on these files is needed.

**Comment by lucenticus** [comment]
@jkotas, I've added verbatim copies, moved them to llvm dir and added hash to this commit. PTAL

**Comment by lucenticus** [comment]
@dotnet-bot test Linux ARM Emulator Cross Release Build please

**Comment by jkotas** [comment]
:+1:

**Comment by richlander** [comment]
LGTM
